src/textual/renderables/sparkline.py

```diff
@@ -33,16 +34,16 @@ def __init__(
         width: int | None,
         min_color: Color = Color.from_rgb(0, 255, 0),
         max_color: Color = Color.from_rgb(255, 0, 0),
-        summary_function: Callable[[list[T]], float] = max,
+        summary_function: SummaryFunction[T] = max,
```

Comment:
The "bug" was here. Not a real bug, but a type-checking bug. If `summary_function` is generic on T, and nothing else uses T, then... I believe the point was to make `Sparkline` generic on T, and everything derives from that T and falls into place. I did that.